### PR TITLE
Oracle: support for binding variables

### PR DESCRIPTION
**Pull request by JirkaChadima**
This is an update of #89.

``` php
$ora = dibi::getConnection('ora');

// functions
$data= $ora->query("select chadima.argtest(%sql, %sql) as r from dual", $ora->getDriver()->bindData('asdfasd3f', SQLT_CHR), $ora->getDriver()->bindData(210, OCI_B_INT))->fetch();

// procedures
$out = $ora->getDriver()->bindOutData(0, OCI_B_INT);
$result = $ora->query("call chadima.argtest_proc(%sql, %sql, %sql)", $ora->getDriver()->bindData('asd1fasdaf', SQLT_CHR), $ora->getDriver()->bindData(212, OCI_B_INT), $out);
var_dump( $ora->getDriver()->getBoundData($out) );
$ora->getDriver()->free(); // this is required for another query when using out parameter

// insert
$key1 = $ora->getDriver()->bindData($rawData, OCI_B_BLOB);
$ora->insert('BINTEST', array(
        'BLOB1%blob' => $key1,
        'IDX' => 3,
     ))
     ->returning('%n INTO %sql', 'BLOB1', $key1)
     ->execute();

// update
$key2 = $ora->getDriver()->bindData($rawText, OCI_B_CLOB);
$ora->update('BINTEST', array(
        'CLOB1%clob' => $key2
     ))
     ->where('IDX = %i', 3)
     ->returning('%n INTO %sql', 'CLOB1', $key2)
     ->execute();

// select
$data = $ora->select('CLOB1')->from('BINTEST')
     ->where('IDX = %i', 3)
     ->fetchSingle();
```

### CODE REVIEW

File: dibi/drivers/DibiOracleDriver.php [hunk 1]
```diff
@@ -97,9 +151,50 @@ public function disconnect()
 	public function query($sql)
 	{
 		$res = oci_parse($this->connection, $sql);
+		if ($this->binds) {
+			set_error_handler(function ($errno, $errstr, $errfile, $errline, $errcontext) use ($sql) {
```
Comment (dg) [inline dibi/drivers/DibiOracleDriver.php]: This error handler is not restored when exception is throw. Is possible to detect error in different way?

Comment (JirkaChadima) [inline dibi/drivers/DibiOracleDriver.php]: You're right, this slipped, I'll fix it. oci_bind_by_name returns boolean which tells you absolutely nothing about the error. In addition it throws warnings when something is not right, that's why I introduced custom error handler.

Comment (JanTvrdik) [inline dibi/drivers/DibiOracleDriver.php]: > which tells you absolutely nothing about the error

Maybe you can use `error_get_last`.

Comment (JirkaChadima) [inline dibi/drivers/DibiOracleDriver.php]: I'll look into it, thanks for the hint.

**Comment by alesculek** [comment]
What is the status of this PR?

**Comment by JirkaChadima** [comment]
@alesculek I no longer have daily access to Oracle nor use PHP. Feel free to take it over, I believe my former colleagues are still using this in production successfully

**Pull request closed without merging**